### ISSUE TITLE
when delete data from mysql database,how synchronous to elasticsearch

### ISSUE THREAD

**Issue by thooo** [jprante/elasticsearch-river-jdbc#511]
After fetching data from JDBC sources for indexing into elasticsearch, I add 1 row data in my database,because I set "schedule" : "0 0-59 0-23 ? * *",one minute later, refresh the elasticsearch cluster , index add 1 document that added in the database indeed,It’s show that the data added in the database synchronized to elasticsearch. It’s OK.
However,when I delete 1 row data from mysql databse,one minute later,refresh the elasticsearch,the index’s document don’t reduce 1,look up the cluster status,”num_docs” show the original document total,and “max_doc” show the number is “num_docs”+1,”deleted_docs” show the number is 1,but the data that delete from databse also can be searched in elasticsearch. Why is it so

**Comment by jprante** [comment]
What is the SQL statement you use to detect deleted documents?

**Comment by thooo** [comment]
MySQL database structure:
table:USER
column:username,password
row1:thooo,thooo
row2:thooo1,thooo1

step 1:execute command
curl -XPUT 'http://localhost:9200/_river/myriver/_meta' -d "{"type":"jdbc","schedule":"0 0-59 0-23 ? * *","jdbc":{"url":"jdbc:mysql://localhost:3306/test","user":"root","password":"root","sql":"select *, username as _id from user","index":"shop","type":"user"}}"
step 2:look up elasticsearch
one minute later,input command "curl -XGET http://localhost:9200/shop/_search?pretty",the hits content:
"hits" : {
    "total" : 2,
    "max_score" : 1.0,
    "hits" : [ {
      "_index" : "shop",
      "_type" : "user",
      "_id" : "thooo1",
      "_score" : 1.0,
      "_source":{"username":"thooo1","password":"thooo1"}
    }, {
      "_index" : "shop",
      "_type" : "user",
      "_id" : "thooo",
      "_score" : 1.0,
      "_source":{"username":"thooo","password":"thooo"}
    } ]
  }
it's OK!
step 3:execute delete SQL in my database
DELETE FROM USER WHERE username='thooo1';at this moment,there is only one row in my datebase
step 4:look up elasticsearch again
one minute later,I refresh elasticsearch and input command:curl -XGET http://localhost:9200/shop/_search?pretty
Originally I thought leaving only one data in elasticsearch ,because execute "select * from user ", there is only one row in my databse.
However,content of hits is as same as content which get in step 2. There are still two datas.
Why not synchronized to elasticsearch after delete data from database.

**Comment by jprante** [comment]
Your statement is `select *, username as _id from user`. This means, all found documents are indexed. 

There is no method to delete document unless you set up an SQL statement such as `select username as _id, 'delete' as _optype from ...`

**Comment by Nab3a** [comment]
Please elaborate further, I have a similar issue.

**Comment by jprante** [comment]
You must detect rows that are deleted in the DB by yourself. This is not automatically done.

**Comment by Nab3a** [comment]
Ok and what about the rows that are Updated, is it automatic?

**Comment by jprante** [comment]
There is no automatic mode. You can set up is a series of SQL statements for the data you want to index or delete in ES, triggered at scheduled intervals.
With ES, you can index daily using index templates and drop older indices, if you want to delete large amount of old data automatically. This is not a feature of JDBC plugin, but of ES.

**Comment by Nab3a** [comment]
Yes I understand that, what I meant to ask was:

I have set "schedule":"0 0/30 * 1/1 * ? *" 

Does this mean that jdbc will have the updated/newly inserted rows normally every 30 minutes??

**Comment by jprante** [comment]
No, this only runs on first day of month. This should work: `0 0/30 * * * ? *`

**Comment by Nab3a** [comment]
Thank you.

**Comment by Nab3a** [comment]
Whenever the schedule is triggered (i.e 30 minutes), the jdbc will get updated rows and new inserted rows? I realized you still haven't answered yes/no 

Sorry for being unclear

**Comment by jprante** [comment]
It depends on your SQL statement.

**Comment by Nab3a** [comment]
my sql is : select column1,column2,column3,column4,column5 from table

**Comment by jprante** [comment]
So, you can see for yourself, updates will not work, because there is no `_id` column.

**Comment by Nab3a** [comment]
I have included it now, what else do I need to do?

**Comment by jprante** [comment]
It would be nice not to capture foreign issues for your questions. You can also ask for help on the Elasticsearch community mailing list.

**Comment by Nab3a** [comment]
You're not going to answer my question at least?

**Comment by jprante** [comment]
Sorry, I really do not know what you "need to do".

**Comment by guyajun** [comment]
Who have solved the problem? Would you please share the solution with me? TKS!
